File: snippet.rb
require 'json'
require 'net/http'

uri = URI("https://api.github.com/gists")
data = { 'files' => { 'snippet' => { 'content' => ARGV[0] }}}

request = Net::HTTP::Post.new(uri.path)
request.body = data.to_json

response = Net::HTTP.start(uri.hostname, uri.port, :use_ssl => true) do |http|
	http.request(request)
end

my_hash = JSON.parse(response.body)
IO.popen('pbcopy', 'w') { |f| f << my_hash["html_url"] }

File: snippet.py
import sys as s
import os as o
import urllib2 as u
import json as j

r = u.Request("https://api.github.com/gists")
r.add_header("Content-type","application/json")
d = {"files":{"snippet":{"content":s.argv[1]}}}
g = j.load(u.urlopen(r,j.dumps(d)))["html_url"]
o.system("echo %s |pbcopy" % g)
